### PR TITLE
Remove type declaration for a module in types package

### PR DESCRIPTION
**Pull request by mmv08**
In the current form, the declaration overwrites types for modules named `@walletconnect/types`, this makes it impossible to properly integrate walletconnect v1 and v2 in the same app, so if you'd try to import types for v2 it will resolve to v1 types.

Required for https://github.com/gnosis/safe-react-apps/issues/178

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -67,6 +67,7 @@
     "lerna": "3.20.2",
     "npm-run-all": "4.1.5",
     "prettier": "1.19.1",
+    "typescript": "^4.4.4",
```
Comment (mmv08) [inline package.json:70]: `npx tsc` in the build step asked me to install `typescript` first

**Comment by pedrouid** [comment]
What I've been recommending instead is that you install using NPM aliases not only for the client but also for it's dependencies.

```
    "@walletconnect/client": "1.6.6",
    "@walletconnect/types": "1.6.6",
    "@walletconnect/utils": "1.6.6",
    "@walletconnect/v2-client": "npm:@walletconnect/client@2.0.0-beta.20",
    "@walletconnect/v2-types": "npm:@walletconnect/types@2.0.0-beta.20",
    "@walletconnect/v2-utils": "npm:@walletconnect/utils@2.0.0-beta.20",
```

I know it's not a perfect solution nor the best developer experience but we intend to improve this 🙏

**Comment by mmv08** [comment]
> What I've been recommending instead is that you install using NPM aliases not only for the client but also for it's dependencies.
> 
> ```
>     "@walletconnect/client": "1.6.6",
>     "@walletconnect/types": "1.6.6",
>     "@walletconnect/utils": "1.6.6",
>     "@walletconnect/v2-client": "npm:@walletconnect/client@2.0.0-beta.20",
>     "@walletconnect/v2-types": "npm:@walletconnect/types@2.0.0-beta.20",
>     "@walletconnect/v2-utils": "npm:@walletconnect/utils@2.0.0-beta.20",
> ```
> 
> I know it's not a perfect solution nor the best developer experience but we intend to improve this 🙏

This is hacky and does not fix the issue. What's wrong with merging my PR that fixes the problem? You don't need to include module declaration for types package because that's literally what the type package is for.

**Comment by mmv08** [comment]
@pedrouid Could you please take another look? I believe the current type packages users `declare` statement wrongly and blocks v2 integration

**Comment by pedrouid** [comment]
Thanks for the PR @mikheevm 🙌  I will handle this on the `next` branch

**Comment by pedrouid** [comment]
Check commit ea0d4e358e5079236de7813887b53a3d88935dd6

**Comment by mmv08** [comment]
thank you!

**Pull request closed without merging**